Rest/Field/TypeValue.php

```diff
@@ -0,0 +1,181 @@
+<?php
+/**
+ * This file is part of the EzSystemsRecommendationBundle package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\RecommendationBundle\Rest\Field;
+
+use eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ConfigResolver;
+use eZ\Bundle\EzPublishCoreBundle\Imagine\AliasGenerator as ImageVariationService;
+use eZ\Publish\API\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\Field;
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use eZ\Publish\Core\FieldType\XmlText\Converter\Html5;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class TypeValue
+{
+    /** @var \Symfony\Component\HttpFoundation\RequestStack */
+    private $request;
+
+    /** @var \eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ConfigResolver */
+    private $configResolver;
+
+    /** @var \eZ\Publish\Core\FieldType\XmlText\Converter\Html5 */
+    private $html5Converter;
+
+    /** @var \eZ\Bundle\EzPublishCoreBundle\Imagine\AliasGenerator */
+    protected $imageVariationService;
+
+    /** @var array */
+    public $fieldDefIdentifiers;
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\RequestStack $request
+     * @param \eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ConfigResolver $configResolver
+     * @param \eZ\Publish\Core\FieldType\XmlText\Converter\Html5 $html5Converter
+     * @param \eZ\Bundle\EzPublishCoreBundle\Imagine\AliasGenerator $imageVariationService
+     */
+    public function __construct(
+        RequestStack $request,
+        ConfigResolver $configResolver,
+        Html5 $html5Converter,
+        ImageVariationService $imageVariationService
+    ) {
+        $this->request = $request;
+        $this->configResolver = $configResolver;
+        $this->html5Converter = $html5Converter;
+        $this->imageVariationService = $imageVariationService;
+    }
+
+    /**
+     * Prepares an array with field type identifiers.
+     * 
+     * @param \eZ\Publish\API\Repository\Values\ContentType\ContentType $contentType
+     */
+    public function setFieldDefs(ContentType $contentType)
+    {
+        foreach ($contentType->fieldDefinitions as $fieldDef) {
+            $this->fieldDefIdentifiers[$fieldDef->identifier] = $fieldDef->fieldTypeIdentifier;
+        }
+    }
+
+    /**
+     * Returns field definition.
+     *
+     * @param string $identifier
+     *
+     * @return mixed
+     */
+    public function getFieldDef($identifier)
+    {
+        return $this->fieldDefIdentifiers[$identifier];
+    }
+
+    /**
+     * Returns field definitions.
+     *
+     * @return mixed
+     */
+    public function getFieldDefs()
```

Comment:
IMO something like: `getFieldDefinitionList()` or `getFieldDefinitionArray()` is better.